driver-mapping/src/main/java/com/datastax/driver/mapping/DefaultPropertyMapper.java

```diff
@@ -0,0 +1,542 @@
+        return this;
+    }
+
+    /**
+     * Sets the {@link PropertyTransienceStrategy property transience strategy} to use.
+     * The default is {@link PropertyTransienceStrategy#OPT_OUT}.
+     *
+     * @param propertyTransienceStrategy the {@link PropertyTransienceStrategy property transience strategy} to use; may not be {@code null}.
+     * @return this {@link DefaultPropertyMapper} instance (to allow for fluent builder pattern).
+     */
+    public DefaultPropertyMapper setPropertyTransienceStrategy(PropertyTransienceStrategy propertyTransienceStrategy) {
+        this.propertyTransienceStrategy = checkNotNull(propertyTransienceStrategy);
+        return this;
+    }
+
+    /**
+     * Sets the {@link HierarchyScanStrategy hierarchy scan strategy} to use.
+     * The default is {@link DefaultHierarchyScanStrategy}.
+     *
+     * @param hierarchyScanStrategy the {@link HierarchyScanStrategy hierarchy scan strategy} to use; may not be {@code null}.
+     * @return this {@link DefaultPropertyMapper} instance (to allow for fluent builder pattern).
+     */
+    public DefaultPropertyMapper setHierarchyScanStrategy(HierarchyScanStrategy hierarchyScanStrategy) {
+        this.hierarchyScanStrategy = checkNotNull(hierarchyScanStrategy);
+        return this;
+    }
+
+    /**
+     * Sets the {@link NamingStrategy naming strategy} to use.
+     * The default is {@link DefaultNamingStrategy}.
+     *
+     * @param namingStrategy the {@link NamingStrategy naming strategy} to use; may not be {@code null}.
+     * @return this {@link DefaultPropertyMapper} instance (to allow for fluent builder pattern).
+     */
+    public DefaultPropertyMapper setNamingStrategy(NamingStrategy namingStrategy) {
+        this.namingStrategy = checkNotNull(namingStrategy);
+        return this;
+    }
+
+    /**
+     * Sets transient property names. This
+     * will completely replace any names already configured for this object.
+     * <p/>
+     * The default set comprises the following property names:
+     * {@code class} and {@code metaClass}.
+     * These properties pertain to the {@link Object} class –
+     * {@code metaClass} being specific to the Groovy language.
+     * <p/>
+     * Property names provided here will always be considered transient;
+     * if a more fine-grained tuning is required, it is also possible
+     * to use the {@link Transient @Transient} annotation
+     * on a specific property.
+     * <p/>
+     * Subclasses can also override {@link #isTransient(String, Field, Method, Method, Map)} to gain
+     * complete control over which properties should be considered transient.
+     *
+     * @param transientPropertyNames a set of property names to exclude from mapping; may not be {@code null}. This
+     *                               will completely replace any names already configured for this object.
+     */
+    public DefaultPropertyMapper setTransientPropertyNames(Set<String> transientPropertyNames) {
```

Comment:
Seems it could be useful for when doing some sort of inheritance in the Java objects when having only one Cassandra table to map to. By providing a list of only the properties that are supposed to be mapped, you can map a table with a Java object, or its child, or its child's childs, etc, without having to explicitly exclude the children's properties via a black list each time. Or when you're denormalizing tables in C*, it could help selecting only some fields to map, without worrying about the other fields in the different denormalized tables, or other fields in the mapped object. Seems useful to me but that's a subjective opinion and we don't really have external feedback requesting this specifically so... ok.